### PR TITLE
CASMPET-6584 Remove obsolete cray-service.postgreSql entries

### PR DESCRIPTION
**Pull request by rnoska-hpe**
CAMPET-6584 When upgrading to CSM 1.5 or later, ensure that we remove obsolete cray-service.postgreSql entries.

# Description

This addition to the CSM upgrade process will ensure that we remove any obsolete instances of cray-service.postgreSql  from the customizations. With the transition to the cray-postgresql base Helm chart, these references are no longer needed. That transition was completed in CSM 1.5 in https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-6402 

# Checklist

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/scripts/upgrade/util/update-customizations.sh [hunk 1]
```diff
@@ -136,6 +136,9 @@ fi
 # Kube-prometheus-stack
 sed -i 's/prometheus-operator/kube-prometheus-stack/g' "$c"
 
+# When upgrading to CSM 1.5 or later, ensure that we remove obsolete cray-service.postgreSql entries (CASMPET-6584).
+yq d -i "$c" 'spec.kubernetes.services.*.cray-service.sqlCluster'
```
Comment (rustydb) [inline upgrade/scripts/upgrade/util/update-customizations.sh:140]: Same comment as yesterday on Ryan's PR.

https://github.com/Cray-HPE/docs-csm/pull/3805/files#r1198206006

```suggestion
yq4 -i 'del(.'spec.kubernetes.services.*.cray-service.sqlCluster)' "$c"
```

I haven't used the wildcard with the new `del()` function, Ryan's PR was just targeting one key when I made the original recommendation.

Comment (rustydb) [reply upgrade/scripts/upgrade/util/update-customizations.sh:140]: More information: https://mikefarah.gitbook.io/yq/upgrading-from-v3

**Review by rustydb — CHANGES_REQUESTED** [review]

File: upgrade/scripts/upgrade/util/update-customizations.sh (hunk 1 above)
Comment (rnoska-hpe) [reply upgrade/scripts/upgrade/util/update-customizations.sh:140]: Yes, that wildcard will work with yq4. I didn't realize that yq4 was on the system earlier when I submitted this. I'll make that change.

Comment (rustydb) [reply upgrade/scripts/upgrade/util/update-customizations.sh:140]: It's been available but not well advertised.
Recently we're taking advocating for using it for new `yq` usages to make the slide to `yq4` easier.

Comment (rnoska-hpe) [reply upgrade/scripts/upgrade/util/update-customizations.sh]: That makes sense. I've pushed a new commit that seems more appropriate based on my review of customizations on Drax (where this issue was discovered). This was tested on Drax and works as expected to update the customizations.

**Review by kburns-hpe — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**